File: java-samples/src/main/java/posts/parthmistry/javasamples/LogLineRegexWithCharCount.java
package posts.parthmistry.javasamples;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.regex.Pattern;

public class LogLineRegexWithCharCount {

    public static void main(String[] args) throws Exception {
        var logFilePath = args[0];

        var charCount = 0L;

        var pattern = Pattern.compile("[\\d]+-[\\d]+-[\\d]+ [\\d]+:[\\d]+:[\\d]+\\.[\\d]+ .* REQUEST on (.*)");

        try (var reader = Files.newBufferedReader(Paths.get(logFilePath))) {
            String line = null;
            while ((line = reader.readLine()) != null) {
                var matcher = pattern.matcher(line);
                if (matcher.matches()) {
                    var matchedGroup = matcher.group(1).toUpperCase();
                    for (int i = 0; i < matchedGroup.length(); i++) {
                        var c = matchedGroup.charAt(i);
                        if (c == 'A' || c == 'E') {
                            charCount += 1;
                        }
                    }
                }
            }
        }

        System.out.println("char count: " + charCount);
    }

}

File: python-samples/log_line_regex_with_char_count.py
import re
import sys

log_file_path = sys.argv[1]

char_count = 0

pattern = re.compile("[\\d]+-[\\d]+-[\\d]+ [\\d]+:[\\d]+:[\\d]+\.[\\d]+ .* REQUEST on (.*)")

with open(log_file_path, 'r') as log_file:
    while line := log_file.readline():
        processed_line = line.rstrip('\n')
        result = pattern.search(processed_line)
        if result is not None:
            matched_group_str = result.groups()[0].upper()
            for c in matched_group_str:
                if c == 'A' or c == 'E':
                    char_count += 1

print("char count:", char_count)
